op-service/log/cli.go

```
@@ -46,70 +47,93 @@ func CLIFlags(envPrefix string) []cli.Flag {
+// LevelFlagValue is a value type for cli.GenericFlag to parse and validate log-level values.
 // Log level: trace, debug, info, warn, error, crit. Capitals are accepted too.
-type LvlFlagValue log.Lvl
+type LevelFlagValue slog.Level
 
-func NewLvlFlagValue(lvl log.Lvl) *LvlFlagValue {
-	return (*LvlFlagValue)(&lvl)
+func NewLevelFlagValue(lvl slog.Level) *LevelFlagValue {
+	return (*LevelFlagValue)(&lvl)
 }
 
-func (fv *LvlFlagValue) Set(value string) error {
+func (fv *LevelFlagValue) Set(value string) error {
 	value = strings.ToLower(value) // ignore case
-	lvl, err := log.LvlFromString(value)
+	lvl, err := LevelFromString(value)
 	if err != nil {
 		return err
 	}
-	*fv = LvlFlagValue(lvl)
+	*fv = LevelFlagValue(lvl)
 	return nil
 }
 
-func (fv LvlFlagValue) String() string {
-	return log.Lvl(fv).String()
+func (fv LevelFlagValue) String() string {
+	return slog.Level(fv).String()
 }
 
-func (fv LvlFlagValue) LogLvl() log.Lvl {
-	return log.Lvl(fv)
+func (fv LevelFlagValue) Level() slog.Level {
+	return slog.Level(fv).Level()
 }
 
-func (fv *LvlFlagValue) Clone() any {
+func (fv *LevelFlagValue) Clone() any {
 	cpy := *fv
 	return &cpy
 }
 
-var _ cliapp.CloneableGeneric = (*LvlFlagValue)(nil)
+// LevelFromString returns the appropriate Level from a string name.
+// Useful for parsing command line args and configuration files.
+// It also converts strings to lowercase.
+func LevelFromString(lvlString string) (slog.Level, error) {
+	lvlString = strings.ToLower(lvlString) // ignore case
+	switch lvlString {
+	case "trace", "trce":
+		return log.LevelTrace, nil
+	case "debug", "dbug":
+		return log.LevelDebug, nil
+	case "info":
+		return log.LevelInfo, nil
+	case "warn":
+		return log.LevelWarn, nil
+	case "error", "eror":
+		return log.LevelError, nil
+	case "crit":
+		return log.LevelCrit, nil
+	default:
+		return log.LevelDebug, fmt.Errorf("unknown level: %v", lvlString)
```

Why the default to `debug` or will the app always exit here?

---

I remember that I just copied this from upstream. 

I think the idea here is to offer the caller the chance to ignore the error and have debug then be the default log level. IMO that's fine and doesn't hurt. But I can describe this behavior in the function comment.